diff.py

```diff
@@ -2050,9 +2078,21 @@ def normalize_stack(row: str, arch: ArchSettings) -> str:
     return re.sub(arch.re_sprel, "addr(sp)", row)
 
 
-def imm_matches_everything(row: str, arch: ArchSettings) -> bool:
-    # (this should probably be arch-specific)
-    return "(." in row
+def imm_matches_everything(field: str, arch: ArchSettings) -> bool:
+    if arch.name == "ppc":
+        if "...data" in field:
+            return True
+
+        field = "".join(field.rsplit("@ha", 1))
+        field = "".join(field.rsplit("@l", 1))
+        field = "".join(field.rsplit("@sda21", 1))
```

Comment:
`field = field.replace("@ha", "").replace("@l", "").replace("@sda21", "")` is easier to read (and kinda ugly that it needs to be done, but understandable...)

if you want it to happen only at the very end of the string, rsplit is not sufficient for that

Reply:
I did it with rsplit so it would only pull from the very end and only once,

I wasn't sure if those C++ mangles symbol names could have this pattern in the middle, but looks like they very easily could.

So right now this works and correctly does not catch C++ mangled ones, but could cause issues if I make this change. So I will hold off. unless you tell me to.

Reply:
also I found if I set it to 
>field.replace("@ha", "").replace("@l", "").replace("@sda21", "")

black reformats it and makes it multi line anyways

Reply:
Could do
```py
parts = field.rsplit("@", 1)
if len(parts) == 2 and parts[1] in {"l", "h", "ha", "sda21"}:
    field = parts[0]
```

Reply:
> (watch out for accidentally mentioning github users when typing `@`s!)

lol!! Whoops

Reply:
Okay I took your latest code